server/src/main/java/org/opensearch/action/search/DeletePitResponse.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.search;
+
+import org.opensearch.action.ActionResponse;
+import org.opensearch.common.ParseField;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.xcontent.ConstructingObjectParser;
+import org.opensearch.common.xcontent.ObjectParser;
+import org.opensearch.common.xcontent.StatusToXContentObject;
+import org.opensearch.common.xcontent.ToXContent;
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.common.xcontent.XContentParser;
+import org.opensearch.rest.RestStatus;
+
+import java.io.IOException;
+
+import static org.opensearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.opensearch.rest.RestStatus.NOT_FOUND;
+import static org.opensearch.rest.RestStatus.OK;
+
+/**
+ * Response class for delete pits flow which returns if the contexts are freed
+ */
+public class DeletePitResponse extends ActionResponse implements StatusToXContentObject {
```

Comment:
The response does not reflect the "bulk" nature of the `DeletePitRequest`: if some PITs were not deleted (for whatever reason), it is not possible to know which ones. That should be provided in response. Alternatively, `DeletePitRequest` could operate on one PIT only: `DELETE _search/point_in_time/{id}`

Reply:
Thought process behind this :
We modeled it similar to clear scroll response since it is quite similar in the functionality.

But we changed the way 'success' of the operation works , 'not found' is also considered success since if user retries and id is deleted already ( not found ) , making the operation failed gives wrong message to the user.

Since we're also in discussions with opensearch dashboards team, in these earlier stages atleast, i think list functionality will be quite useful, so giving option only to delete one id might not work.

Reply:
How about adding failed pitIds to the response, so that user can retry deleting those IDs ?

Reply:
I agree with @reta. The response to any bulk-style API needs to be able to communicate the details of a partial failure (unless you can fully guarantee all-or-nothing semantics of the batch, which I don't think is possible here). The behavior you described to treat deleting a non-existent resource as a successful operation seems reasonable to me, but there are likely other failure scenarios where knowing the details of a partial failure is required.

Reply:
Changed the response to capture result for each PIT ID